packages/opencensus-core/src/trace/model/span-base.ts

```diff
@@ -133,7 +137,17 @@ export abstract class SpanBase implements types.Span {
    * @param value The result of an operation.
    */
   addAttribute(key: string, value: string|number|boolean) {
-    this.attributes[key] = value;
+    this.totalRecordedAttributes++;
+    if (Object.keys(this.attributes.attributeMap).length >=
+        this.activeTraceParams.numberOfAttributesPerSpan) {
+      const attributeKeyToDelete =
+          Object.keys(this.attributes.attributeMap).shift();
+      delete this.attributes.attributeMap[attributeKeyToDelete];
+    }
+
+    this.attributes.attributeMap[key] = value;
+    this.attributes.droppedAttributesCount = this.totalRecordedAttributes -
```

Comment:
Would it work to just increment `this.attributes.droppedAttributesCount` in the above `if` block?

Reply:
Make Sense, Done. 👍